service/src/main/java/com/derpgroup/derpwizard/dao/impl/H2EmbeddedAccountLinkingDAO.java

```diff
@@ -0,0 +1,407 @@
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public void expireGrantedToken(String token) {
+    String accessTokenDelete = "DELETE FROM Authorization WHERE token = ?;";
+
+    ArrayList<String> parameters = new ArrayList<String>();
+    parameters.add(token);
+    executeStatement(accessTokenDelete, parameters);
+  }
+  
+  protected PreparedStatement prepStatement(PreparedStatement statement, ArrayList<? extends Object> parameters) throws SQLException{
+    if(parameters != null){
+      for(int i = 0; i < parameters.size(); i++){
+        statement.setObject(i + 1, parameters.get(i));
+      }
+    }
+    return statement;
+  }
+  
+  @Override
+  public ExternalAccountLink createAccountLink(ExternalAccountLink link){
+    ArrayList<Object> parameters = new ArrayList<Object>();
+    parameters.add(link.getUserId());
+    parameters.add(link.getExternalUserId());
+    parameters.add(link.getExternalSystemName());
+    parameters.add(link.getAuthToken());
+    parameters.add(link.getRefreshToken());
+    String createAccountLink = "MERGE INTO AccountLink(userId, externalUserId, externalSystemName, externalSystemToken, externalSystemRefreshToken)"
+        + " KEY(userId, externalUserId, externalSystemName) VALUES(?,?,?,?,?)";
+    //Should externalSystemToken be a different table with a FK relationship to this one?
+
+    executeStatement(createAccountLink, parameters);
+    
+    return getAccountLinkByUserIdAndExternalSystemName(link.getUserId(), link.getExternalSystemName());
+  }
+  
+  @Override
+  public ExternalAccountLink getAccountLinkByUserIdAndExternalSystemName(String userId, String externalSystemName){
+    ArrayList<Object> parameters = new ArrayList<Object>();
+    parameters.add(userId);
+    parameters.add(externalSystemName);
+    String createAccountLink = "SELECT userId, externalUserId, externalSystemName, externalSystemToken, externalSystemRefreshToken"
+        + " FROM AccountLink"
+        + " WHERE userId = ?"
+        + " AND externalSystemName = ?";
+
+    ResultSet response;
+    response = executeQuery(createAccountLink, parameters);
+    try{
+      if(!response.next()){
+        return null;
+      }
+      
+      return buildAccountLink(response);
+    }catch(SQLException e){
+      e.printStackTrace();
+      return null;
+    }
+  }
+  
+  @Override
+  public ExternalAccountLink getAccountLinkByExternalUserIdAndExternalSystemName(String externalUserId, String externalSystemName){
+    ArrayList<Object> parameters = new ArrayList<Object>();
+    parameters.add(externalUserId);
+    parameters.add(externalSystemName);
+    String createAccountLink = "SELECT userId, externalUserId, externalSystemName, externalSystemToken, externalSystemRefreshToken"
+        + " FROM AccountLink"
+        + " WHERE externalUserId = ?"
+        + " AND externalSystemName = ?";
+
+    ResultSet response;
+    response = executeQuery(createAccountLink, parameters);
```

Comment:
Is this safe? Does it need a try/catch around it? And can it return null (does it need a null check?)